http/org.wso2.carbon.transport.http.netty/src/test/java/org/wso2/carbon/transport/http/netty/util/TestUtil.java

```diff
@@ -175,6 +175,24 @@ public static HttpsServer startHttpsServer(int port, String message, String cont
         return httpServer;
     }
 
+    public static HttpServer startHTTPServerForRedirect(int port, String message, String contentType, int
+            responseCode, String location, int delay) {
+        HttpServer httpServer = new HttpServer(port);
+        CountDownLatch latch = new CountDownLatch(1);
+        ServerThread serverThread = new ServerThread(latch, httpServer);
+        try {
+            serverThread.start();
+            latch.await();
+            httpServer.setMessage(message, contentType);
+            httpServer.setResponseCode(responseCode);
+            httpServer.setLocation(location);
+            httpServer.setDelay(delay);
+        } catch (Exception e) {
+            log.error("Thread Interrupted while sleeping ", e);
```

Comment:
This will not only grab interrupted exceptions ? any reason for using generic exceptions here ?

Reply:
Changed to capture only InterruptedException